crates/test-framework/src/spicetest/consistency/mod.rs

```diff
@@ -0,0 +1,228 @@
+            .clone();
+
+        let start_time = SystemTime::now();
+        match self {
+            HttpConsistencyComponent::Model { model, .. } => {
+                let req: CreateChatCompletionRequest =
+                    match serde_json::from_str::<CreateChatCompletionRequest>(payload) {
+                        Ok(mut req) => {
+                            // Ensure the model is overriden.
+                            req.model = model.clone();
+                            req
+                        }
+                        Err(_) => CreateChatCompletionRequestArgs::default()
+                            .model(model.clone())
+                            .messages(vec![ChatCompletionRequestMessage::User(
+                                ChatCompletionRequestUserMessageArgs::default()
+                                    .content(payload.to_string())
+                                    .build()
+                                    .expect("failed to build user message"),
+                            )])
+                            .build()
+                            .expect("failed to build model request"),
+                    };
+                let _ = c.chat().create(req).await?;
+            }
+            HttpConsistencyComponent::Embedding { embedding, .. } => {
+                let req: CreateEmbeddingRequest =
+                    match serde_json::from_str::<CreateEmbeddingRequest>(payload) {
+                        Ok(mut req) => {
+                            // Ensure the model is overriden.
+                            req.model = embedding.clone();
+                            req
+                        }
+                        Err(_) => CreateEmbeddingRequest {
+                            model: embedding.clone(),
+                            input: async_openai::types::EmbeddingInput::String(payload.to_string()),
+                            encoding_format: Some(EncodingFormat::Float),
+                            user: None,
+                            dimensions: None,
+                        },
+                    };
+                c.embeddings().create(req).await?;
+            }
+        }
+        Ok(start_time.elapsed()?)
+    }
+}
+
+pub enum ConsistencyState {
+    NotStarted,
+    Running { worker_handles: Vec<WorkerHandle> },
+    Completed { result: ConsistencyWorkerResult },
+}
+
+pub struct ConsistencySpiceTest {
+    start_time: Option<SystemTime>,
+    spiced_instance: SpicedInstance,
+    config: ConsistencyConfig,
+    state: ConsistencyState,
+}
+
+impl ConsistencySpiceTest {
+    #[must_use]
+    pub fn new(spiced_instance: SpicedInstance, config: ConsistencyConfig) -> Self {
+        Self {
+            start_time: None,
+            spiced_instance,
+            config,
+            state: ConsistencyState::NotStarted,
+        }
+    }
+
+    pub async fn start(self) -> anyhow::Result<ConsistencySpiceTest> {
+        if !matches!(self.state, ConsistencyState::NotStarted) {
```

Comment:
If you look at the pattern of the SpiceTest, it is parameterized on the state (i.e. `struct SpiceTest<S: TestState>`) which means you can define functions to only exist when its in a particular state, which uses the type system to enforce this check at compile time rather than at runtime